tests/integration_tests/dashboards/filter_state/api_tests.py

```diff
@@ -0,0 +1,111 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# isort:skip_file
+import pytest
+import json
+from superset import security_manager, app
+from superset.models.dashboard import Dashboard
+from superset.extensions import cache_manager
+from tests.integration_tests.base_tests import SupersetTestCase
+from tests.integration_tests.test_app import app
+from sqlalchemy.orm import Session
+
+dashboardId = 1
+value = "test"
+
+
+class FilterStateTests(SupersetTestCase):
```

Comment:
One last thing - can you add some tests that validate the user perms are working as expected?

I.e. make some tests that verify that a user can't access a dashboard's KV's if the same user doesn't have access to the target dashboard.

Reply:
Done